crates/sui-graphql-rpc/src/types/event.rs

```diff
@@ -107,8 +107,14 @@ impl Event {
             return Ok(None);
         }
 
+        let checkpoint_viewed_at = self
```

Comment:
Noting that this also shouldn't be the event's tx's checkpoint sequence number, but the checkpoint that the data was viewed at. (But I suspect your plan is to clean this up in the follow-up, and similar for the `BalanceChange` case?)

Reply:
Yep! https://github.com/MystenLabs/sui/pull/15960